File: lab1/object/Driver.cs
public class Driver : User
{
    public string VehicleType { get; set; }
    public bool Availability { get; set; }

    public void AcceptRide(Trip trip)
    {
        // Logic for driver to accept a ride
        Console.WriteLine("Driver has accepted the ride.");
    }

    public void UpdateLocation(string location)
    {
        this.Location = location;
        // Logic to update driver's location
    }

    public void RateRider(Rider rider, int rating)
    {
        // Logic to rate the rider
        rider.Rating = (rider.Rating + rating) / 2; // Averaging ratings
    }

    public void StartTrip(Trip trip)
    {
        trip.Status = "Started";
        Console.WriteLine("Trip has started.");
    }
}

File: lab02_stratergy_pattern/src/Driver.java

// Driver class
class Driver {
    private String name;
    private String location;

    public Driver(String name, String location) {
        this.name = name;
        this.location = location;
    }

    public String getName() {
        return name;
    }

    public String getLocation() {
        return location;
    }

    public void acceptRide() {
        System.out.println(name + " has accepted the ride.");
    }
}